### PR TITLE
Do not send rules reported that are disabled for the cluster

### PR DESCRIPTION
**Pull request by joselsegura**
# Description

The rules included for each report in the endpoint for a single cluster includes the disabled rules too. The report should not include those rules.

Fixes #[insights-results-aggregator#1128](https://github.com/RedHatInsights/insights-results-aggregator/issues/1128)

## Type of change
- Refactor (refactoring code, removing useless files)

## Testing steps

Regular CI

## Checklist
* [x] `make before_commit` passes
* [x] updated documentation wherever necessary
* [x] added or modified tests if necessary
* [x] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

**Review by tisnik — APPROVED** [review]
Is it THAT easy? :) LGTM of course

**Comment by joselsegura** [comment]
Well, I need to double check because I think we are still storing the "disabled" flag and user votes per user and not per cluster, so I want to discuss it tomorrow when the crew is back to work :-)

**Review by Bee-lee — CHANGES_REQUESTED** [review]
If we merge this, we would lose the functionality in OCM to re-enable the rule.
The `disabled` flag is checked on OCM frontend and such disabled rules are then put in the disabled health checks section, from which the user has the ability to re-enable the rule.

**Comment by joselsegura** [comment]
> If we merge this, we would lose the functionality in OCM to re-enable the rule.
> The `disabled` flag is checked on OCM frontend and such disabled rules are then put in the disabled health checks section, from which the user has the ability to re-enable the rule.

Very good point @Bee-lee I didn't think on the OCM part at all. I will review with somebody from presentation team to be sure to don't broke anything

**Review by Bee-lee — APPROVED** [review]
LGTM!

**Review by dpensi — APPROVED** [review]

File: server/server.go [hunk 1]
```diff
@@ -825,3 +772,34 @@ func (server HTTPServer) getOverviewPerCluster(
...
+) {
+	filteredRules = []proxy_types.RuleWithContentResponse{}
+	noContentRules = 0
+
+	for _, aggregatorRule := range aggregatorReport {
+		if aggregatorRule.Disabled && filterDisabled {
+			continue
+		}
+
+		rule, successful, filtered := content.FetchRuleContent(aggregatorRule, filterOSD)
```
Comment (tisnik) [inline server/server.go:792]: are we blindly "forget" rules w/o content there?

File: server/server_test.go [hunk 2]
```diff
@@ -36,7 +36,7 @@ import (
 )
 
 const (
-	testTimeout            = 10 * time.Second
+	testTimeout            = 5 * time.Second
```
Comment (tisnik) [inline server/server_test.go:39]: it won't work in all times on CI, really ;)

**Review by tisnik — APPROVED** [review]
LGTM

File: server/server.go (hunk 1 above)
Comment (joselsegura) [reply server/server.go:792]: The next `if` block checks it and increase the `noContentRules` variable, which is one of the returned ones. If I'm not mistaken, the code behaves exactly as the previous version (which probably needs a refactor, maybe as a different task)

File: server/server_test.go (hunk 2 above)
Comment (joselsegura) [reply server/server_test.go:39]: Ouch, I forgot to increase it again. I'll remove this change

**Pull request merged**